restli-server/src/main/java/com/linkedin/restli/internal/server/response/ResponseUtils.java

```diff
@@ -84,6 +95,209 @@ static Object translateCanonicalKeyToAlternativeKeyIfNeeded(Object canonicalKey,
     }
   }
 
+  public static DataMap fillInDefaultOnRecord(RecordDataSchema schema, DataMap dataMap)
+  {
+    DataMap dataWithDefault = new DataMap(DataMapBuilder.getOptimumHashMapCapacityFromSize(dataMap.size()));
+    dataWithDefault.putAll(dataMap);
+    for (RecordDataSchema.Field field : schema.getFields())
+    {
+      DataSchema dataFieldSchema = field.getType();
+      if (dataFieldSchema.getType() == DataSchema.Type.RECORD)
+      {
+        if (dataMap.containsKey(field.getName()) || field.getDefault() != null)
```

Comment:
this condition seems common and can be moved out of data schema switch construct